resource "routeros_queue_tree" crashes when refreshing state

version: 1.99.0
`# QUEUE TYPES

resource "routeros_queue_type" "cake-vdsl-download-magio-proaut" {
  name                  = "cake-vdsl-download-magio-proaut"
  kind                  = "cake"
  cake_ack_filter       = "none"
  cake_diffserv         = "diffserv4"
  cake_flowmode         = "triple-isolate" # "dual-dsthost"
  cake_nat              = true
  cake_overhead_scheme  = "pppoe-ptm"
  cake_atm              = "ptm"
  cake_rtt_scheme       = "regional"
  cake_autorate_ingress = false
}

resource "routeros_queue_type" "cake-vdsl-upload-magio-proaut" {
  name                  = "cake-vdsl-upload-magio-proaut"
  kind                  = "cake"
  cake_ack_filter       = "filter"
  cake_diffserv         = "diffserv4"
  cake_flowmode         = "triple-isolate" # "dual-srchost"
  cake_nat              = true
  cake_overhead_scheme  = "pppoe-ptm"
  cake_atm              = "ptm"
  cake_rtt_scheme       = "regional"
  cake_autorate_ingress = false
}

# QUEUE Trees - this crashes while parsing the bandwidth values on refresh so it can not be used yet

resource "routeros_queue_tree" "download-magio-proaut" {
  name        = "download-magio-proaut"
  parent      = "global"
  max_limit   = "46500k"
  packet_mark = ["magio-proaut-inbound"]
  queue       = routeros_queue_type.cake-vdsl-download-magio-proaut.name
}`

2026-01-22T19:28:45.368+0100 [ERROR] **provider.terraform-provider-routeros_v1.99.0: panic: [BitsEqual] parse 'new' value error: bits: unknown unit "107" in value "46500k" 2026-01-22T19:28:45.368+0100 [ERROR] provider.terraform-provider-routeros_v1.99.0 2026-01-22T19:28:45.368+0100 [ERROR] provider.terraform-provider-routeros_v1.99.0: goroutine 905 [running]:**

--- a/routeros/parse_duration.go
+++ b/routeros/parse_duration.go
@@ -23,6 +23,16 @@ func ParseDuration(s string, baseUnits time.Duration) (time.Duration, error) {
 	orig := s
 	var d int64
 
+	// 1d 00:00:00
+	if ss := strings.SplitN(s, " ", 2); len(ss) > 1 {
+		res, err := ParseDuration(ss[1], baseUnits)
+		if err != nil {
+			return 0, err
+		}
+		d += int64(res)
+		s = ss[0]
+	}
+
 	// Special case: if all that is left is "0", this is zero.
 	if s == "0" {
 		return 0, nil

--- a/routeros/parse_duration_test.go
+++ b/routeros/parse_duration_test.go
@@ -23,6 +23,8 @@ func TestParseDuration(t *testing.T) {
 		{name: "163w4d9h", args: args{"163w4d9h"}, want: time.Duration(98960400 * time.Second), wantErr: false},
 		{name: "27489h", args: args{"27489h"}, want: time.Duration(98960400 * time.Second), wantErr: false},
 		{name: "120ms", args: args{"0.12"}, want: time.Duration(120 * time.Millisecond), wantErr: false},
+		{name: "1d 00:00:00", args: args{"1d 00:00:00"}, want: time.Duration(24 * time.Hour), wantErr: false},
+		{name: "2w 1d 00:00:00", args: args{"2w 1d 00:00:00"}, want: time.Duration(24 * 15 * time.Hour), wantErr: false},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

